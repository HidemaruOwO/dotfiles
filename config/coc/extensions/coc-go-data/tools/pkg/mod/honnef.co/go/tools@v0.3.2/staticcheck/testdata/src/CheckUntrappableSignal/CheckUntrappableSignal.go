package main

import (
	"os"
	"os/signal"
	"syscall"
)

func fn() {
	c := make(chan os.Signal, 1)
	signal.Notify(c, os.Interrupt)
	signal.Ignore(os.Signal(syscall.SIGKILL)) //@ diag(`cannot be trapped`)
	signal.Ignore(os.Kill)                    //@ diag(`cannot be trapped`)
	signal.Notify(c, os.Kill)                 //@ diag(`cannot be trapped`)
	signal.Reset(os.Kill)                     //@ diag(`cannot be trapped`)
	signal.Ignore(syscall.SIGKILL)            //@ diag(`cannot be trapped`)
	signal.Notify(c, syscall.SIGKILL)         //@ diag(`cannot be trapped`)
	signal.Reset(syscall.SIGKILL)             //@ diag(`cannot be trapped`)
}
