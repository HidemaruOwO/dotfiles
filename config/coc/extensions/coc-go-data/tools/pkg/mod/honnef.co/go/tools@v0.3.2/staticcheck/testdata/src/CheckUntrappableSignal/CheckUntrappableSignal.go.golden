-- remove syscall.SIGKILL from list of arguments --
package main

import (
	"os"
	"os/signal"
	"syscall"
)

func fn() {
	c := make(chan os.Signal, 1)
	signal.Notify(c, os.Interrupt)
	signal.Ignore()           //@ diag(`cannot be trapped`)
	signal.Ignore(os.Kill)    //@ diag(`cannot be trapped`)
	signal.Notify(c, os.Kill) //@ diag(`cannot be trapped`)
	signal.Reset(os.Kill)     //@ diag(`cannot be trapped`)
	signal.Ignore()           //@ diag(`cannot be trapped`)
	signal.Notify(c)          //@ diag(`cannot be trapped`)
	signal.Reset()            //@ diag(`cannot be trapped`)
}

-- remove os.Kill from list of arguments --
package main

import (
	"os"
	"os/signal"
	"syscall"
)

func fn() {
	c := make(chan os.Signal, 1)
	signal.Notify(c, os.Interrupt)
	signal.Ignore(os.Signal(syscall.SIGKILL)) //@ diag(`cannot be trapped`)
	signal.Ignore()                           //@ diag(`cannot be trapped`)
	signal.Notify(c)                          //@ diag(`cannot be trapped`)
	signal.Reset()                            //@ diag(`cannot be trapped`)
	signal.Ignore(syscall.SIGKILL)            //@ diag(`cannot be trapped`)
	signal.Notify(c, syscall.SIGKILL)         //@ diag(`cannot be trapped`)
	signal.Reset(syscall.SIGKILL)             //@ diag(`cannot be trapped`)
}

-- use syscall.SIGTERM instead of syscall.SIGKILL --
package main

import (
	"os"
	"os/signal"
	"syscall"
)

func fn() {
	c := make(chan os.Signal, 1)
	signal.Notify(c, os.Interrupt)
	signal.Ignore(syscall.SIGTERM)    //@ diag(`cannot be trapped`)
	signal.Ignore(os.Kill)            //@ diag(`cannot be trapped`)
	signal.Notify(c, os.Kill)         //@ diag(`cannot be trapped`)
	signal.Reset(os.Kill)             //@ diag(`cannot be trapped`)
	signal.Ignore(syscall.SIGTERM)    //@ diag(`cannot be trapped`)
	signal.Notify(c, syscall.SIGTERM) //@ diag(`cannot be trapped`)
	signal.Reset(syscall.SIGTERM)     //@ diag(`cannot be trapped`)
}

-- use syscall.SIGTERM instead of os.Kill --
package main

import (
	"os"
	"os/signal"
	"syscall"
)

func fn() {
	c := make(chan os.Signal, 1)
	signal.Notify(c, os.Interrupt)
	signal.Ignore(os.Signal(syscall.SIGKILL)) //@ diag(`cannot be trapped`)
	signal.Ignore(syscall.SIGTERM)            //@ diag(`cannot be trapped`)
	signal.Notify(c, syscall.SIGTERM)         //@ diag(`cannot be trapped`)
	signal.Reset(syscall.SIGTERM)             //@ diag(`cannot be trapped`)
	signal.Ignore(syscall.SIGKILL)            //@ diag(`cannot be trapped`)
	signal.Notify(c, syscall.SIGKILL)         //@ diag(`cannot be trapped`)
	signal.Reset(syscall.SIGKILL)             //@ diag(`cannot be trapped`)
}
